pyro/contrib/forecast/evaluate.py

```diff
@@ -173,3 +189,38 @@ def backtest(data, covariates, model_fn, guide_fn=None, *,
         del pred
 
     return results
+
+
+def _warm_start_param_store(t0, t1, t0_old, t1_old):
+    """
+    Helper to update params in the param store in-place as the backtesting
+    window changes. New blocks are initialized to zero in unconstrained space,
+    equivalent to :func:`pyro.infer.autoguide.initialization.init_to_feasible`.
+    """
+    assert t0_old <= t0
+    assert t1_old <= t1
+    assert (t0_old < t0) or (t1_old < t1)
+    store = pyro.get_param_store()
+    for name, constrained_param in store.items():
+        param = constrained_param.unconstrained()
+        time_dim = getattr(param, "_pyro_time_dim", None)
+        if time_dim is None:
+            continue
+        assert time_dim < 0
+        time_dim += constrained_param.dim() - param.dim()
+        with torch.no_grad():
+
+            # Truncate params that have just left the window.
+            if t0 != t0_old:
+                dots = (slice(None),) * (param.dim() + time_dim)
+                param = param[dots + (slice(t0_old - t0, None),)]
```

Comment:
I think `t0_old - t0` should be `t0 - t0_old`. In addition, you might not need the condition `if t0 != t0_old`.